storage/src/vespa/storage/persistence/persistencethread.cpp

```diff
@@ -894,115 +894,24 @@ bool hasBucketInfo(const api::StorageMessage& msg)
-                if (dynamic_cast<api::PutReply*>(replies[i]->getReply().get()))
-                {
-                    ++nputs;
-                } else if (dynamic_cast<api::RemoveReply*>(
-                                replies[i]->getReply().get()))
-                {
-                    ++nremoves;
-                } else {
-                    ++nother;
-                }
-            }
-            LOG_BUCKET_OPERATION(
-                    bucket.getBucketId(),
-                    vespalib::make_string(
-                            "flushing %zu operations (%zu puts, %zu removes, "
-                            "%zu other)",
-                            replies.size(), nputs, nremoves, nother));
-        }
-#endif
-        spi::Bucket b(bucket, spi::PartitionId(_env._partition));
-        // Flush is not used for anything currentlu, and the context is not correct either when batching is done
-        // So just faking it here.
-        spi::Context dummyContext(documentapi::LoadType::DEFAULT, 0, 0);
-        spi::Result result = _spi.flush(b, dummyContext);
-        uint32_t errorCode = _env.convertErrorCode(result);
-        if (errorCode != 0) {
-            for (uint32_t i = 0; i < replies.size(); ++i) {
-                replies[i]->getReply()->setResult(api::ReturnCode((api::ReturnCode::Result)errorCode, result.getErrorMessage()));
-            }
-        }
-    } catch (std::exception& e) {
-        for (uint32_t i = 0; i < replies.size(); ++i) {
-            replies[i]->getReply()->setResult(api::ReturnCode(api::ReturnCode::INTERNAL_FAILURE, e.what()));
-        }
-    }
-
-    for (uint32_t i = 0; i < replies.size(); ++i) {
-        LOG(spam, "Sending reply up (batched): %s %" PRIu64,
-            replies[i]->getReply()->toString().c_str(), replies[i]->getReply()->getMsgId());
-        _env._fileStorHandler.sendReply(replies[i]->getReply());
-    }
-
-    replies.clear();
-}
-
-void PersistenceThread::processMessages(FileStorHandler::LockedMessage & lock)
-{
+void PersistenceThread::processMessages(FileStorHandler::LockedMessage & lock) {
     std::vector<MessageTracker::UP> trackers;
     document::Bucket bucket = lock.first->getBucket();
 
-    while (lock.second) {
-        LOG(debug, "Inside while loop %d, nodeIndex %d, ptr=%p", _env._partition, _env._nodeIndex, lock.second.get());
-        std::shared_ptr<api::StorageMessage> msg(lock.second);
-        bool batchable = isBatchable(*msg);
-
-        // If the next operation wasn't batchable, we should flush
-        // everything that came before.
-        if (!batchable) {
-            flushAllReplies(bucket, trackers);
-        }
-
-        std::unique_ptr<MessageTracker> tracker = processMessage(*msg);
-        if (!tracker || !tracker->getReply()) {
-            // Was a reply
-            break;
-        }
+    LOG(debug, "Inside while loop %d, nodeIndex %d, ptr=%p", _env._partition, _env._nodeIndex, lock.second.get());
```

Comment:
Consider updating log message to no longer talk about looping

Reply:
Fixed